### DISCUSSION TITLE
Why .Net Framework 4.8 WPF app startup time is two times faster than dotnet core?

### DISCUSSION THREAD

**Discussion by ekalchev** [dotnet/runtime#78120] [General]
So I've read about .net 7 performance improvements and I decided to try what is the improvement of the startup time of WPF app ran in net48 vs net6.0-windows and net7.0-windows. This is the test code https://github.com/ekalchev/WpfAppStartupTime

Results on my machines are
200ms for net48
430ms for net6 and net7

I tried to compile some heavy WPF app to see if what will be the difference. I took https://github.com/fluentribbon/Fluent.Ribbon and I can see about 200-300ms longer startup times in the net6 and net7. I wonder why after all the fuss with performance improvements the startup time is significantly slower. Can I do something to improve this?

**Comment by teo-tsirpanis** [comment]
You might be interested in [ReadyToRun](https://learn.microsoft.com/en-us/dotnet/core/deploying/ready-to-run).
